### PR TITLE
fix(i18n): 修正多语言文件编码与单文件发布

### PR DESCRIPTION
**Pull request by kimdiego2098**
# Description

修正多语言文件编码与单文件发布

## Issue reference

fixes BlazorComponent/Masa.Blazor#1074

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by kimdiego2098** [comment]
@doddgu 大佬审核一下

**Comment by doddgu** [comment]
@capdiem PTAL

File: src/Component/BlazorComponent/I18n/I18nReader.cs [hunk 1]
```diff
@@ -11,9 +11,10 @@ public static class I18nReader
         AllowTrailingCommas = true
     };
 
-    public static Dictionary<string, string> Read(string jsonData)
+    public static Dictionary<string, string> Read(string jsonData, Encoding encoding = null)
```
Comment (capdiem) [inline src/Component/BlazorComponent/I18n/I18nReader.cs:14]:
```suggestion
    public static Dictionary<string, string> Read(string jsonData, Encoding? encoding = null)
```

File: src/Component/BlazorComponent/I18n/I18nServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -39,19 +40,19 @@ public static IBlazorComponentBuilder AddI18n(this IBlazorComponentBuilder build
     /// <param name="builder"></param>
     /// <param name="localeDirectory">i18n resource folder physical path,i18n resource file name will be used as culture name</param>
     /// <returns></returns>
-    public static IBlazorComponentBuilder AddI18nForServer(this IBlazorComponentBuilder builder, string localeDirectory)
+    public static IBlazorComponentBuilder AddI18nForServer(this IBlazorComponentBuilder builder, string localeDirectory, Encoding encoding = null)
```
Comment (capdiem) [inline src/Component/BlazorComponent/I18n/I18nServiceCollectionExtensions.cs:43]:
```suggestion
    public static IBlazorComponentBuilder AddI18nForServer(this IBlazorComponentBuilder builder, string localeDirectory, Encoding? encoding = null)
```

File: src/Component/BlazorComponent/I18n/I18nReader.cs [hunk 3]
```diff
@@ -11,9 +11,10 @@ public static class I18nReader
...
     {
-        var reader = new Utf8JsonReader(Encoding.Default.GetBytes(jsonData), _jsonReaderOptions);
+        if (encoding == null) encoding = Encoding.UTF8;
```
Comment (capdiem) [inline src/Component/BlazorComponent/I18n/I18nReader.cs:16]:
```suggestion
        encoding ??= Encoding.UTF8;
```

File: src/Component/BlazorComponent/I18n/I18nServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -73,7 +74,7 @@ public static IBlazorComponentBuilder AddI18nForServer(this IBlazorComponentBuil
         return builder;
     }
 
-    public static async Task<IBlazorComponentBuilder> AddI18nForWasmAsync(this IBlazorComponentBuilder builder, string localesDirectoryApi)
+    public static async Task<IBlazorComponentBuilder> AddI18nForWasmAsync(this IBlazorComponentBuilder builder, string localesDirectoryApi, Encoding encoding = null)
```
Comment (capdiem) [inline src/Component/BlazorComponent/I18n/I18nServiceCollectionExtensions.cs:77]:
```suggestion
    public static async Task<IBlazorComponentBuilder> AddI18nForWasmAsync(this IBlazorComponentBuilder builder, string localesDirectoryApi, Encoding? encoding = null)
```

File: src/Component/BlazorComponent/I18n/I18nServiceCollectionExtensions.cs [hunk 5]
```diff
@@ -97,7 +98,7 @@ public static async Task<IBlazorComponentBuilder> AddI18nForWasmAsync(this IBlaz
         return builder;
     }
 
-    private static void AddI18nFromPath(string path)
+    private static void AddI18nFromPath(string path, Encoding encoding = null)
```
Comment (capdiem) [inline src/Component/BlazorComponent/I18n/I18nServiceCollectionExtensions.cs:101]:
```suggestion
    private static void AddI18nFromPath(string path, Encoding? encoding = null)
```

**Review by capdiem — CHANGES_REQUESTED** [review]

**Comment by kimdiego2098** [comment]
@capdiem 流程不大熟悉，是不是已经改正完了

**Comment by capdiem** [comment]
@kimdiego2098 解决完了

**Review by capdiem — APPROVED** [review]
LGTM

**Pull request merged**